ecommerce/extensions/payment/processors/liqpay.py

```diff
@@ -0,0 +1,212 @@
+""" LiqPay payment processing. """
+import hashlib
+import json
+import logging
+from base64 import b64decode, b64encode
+from decimal import Decimal
+
+import requests
+from django.urls import reverse
+from oscar.apps.payment.exceptions import GatewayError
+from urllib.parse import urljoin
+
+from ecommerce.core.url_utils import get_ecommerce_url
+from ecommerce.extensions.payment.exceptions import DuplicateReferenceNumber, InvalidSignatureError
+from ecommerce.extensions.payment.processors import BasePaymentProcessor, HandledProcessorResponse
+
+
+logger = logging.getLogger(__name__)
+
+
+class Liqpay(BasePaymentProcessor):
+    """
+    Constructs a new instance of the Liqpay processor (October 2019)
+
+    For reference, see https://www.liqpay.ua/documentation/en/api/home/.
+    """
+
+    NAME = 'liqpay'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the Liqpay processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor.
+        """
+        super(Liqpay, self).__init__(site)
+        # NOTE(smandziuk):
+        # - To enable Liqpay sandbox mode add to config entry - 'sandbox_mode': 1,
+        # - To change version of liqpay API add to config entry - 'version' (on october 2019 api version = '3')
+        self.sandbox_mode = self.configuration.get('sandbox_mode', 0)
+        self.version = self.configuration.get('version', '3')
+
+    @property
+    def cancel_url(self):
+        return get_ecommerce_url(self.configuration['cancel_checkout_path'])
```

Comment:
`cancel_checkout_path` is always exists in the `self.configuration`?

Reply:
Yes, it's always exists